src/test/java/org/broadinstitute/barclay/help/DocumentationGenerationIntegrationTest.java

```diff
@@ -42,38 +42,46 @@
         // set the doclet class
         docArgList.add("-doclet");
         docArgList.add(docletClass.getName());
+        // set the index and output extension
+        docArgList.add("-index-file-extension");
+        docArgList.add(indexFileExtension);
+        docArgList.add("-output-file-extension");
+        docArgList.add(outputFileExtension);
         return docArgList.toArray(new String[]{});
     }
 
     @DataProvider
     public Object[][] getDocGenTestParams() {
         return new Object[][] {
                 // default doclet and templates
-                {HelpDoclet.class, new File("src/main/resources/org/broadinstitute/barclay/helpTemplates/"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/HelpDoclet")},
+                {HelpDoclet.class, new File("src/main/resources/org/broadinstitute/barclay/helpTemplates/"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/HelpDoclet"), "html", "html"},
                 // custom doclet and templates
-                {TestDoclet.class, new File("src/test/resources/org/broadinstitute/barclay/help/templates/TestDoclet"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/TestDoclet")}
+                {TestDoclet.class, new File("src/test/resources/org/broadinstitute/barclay/help/templates/TestDoclet"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/TestDoclet"), "html", "html"},
+                // custom templates with different extension
+                {HelpDoclet.class, new File("src/test/resources/org/broadinstitute/barclay/help/templates/markdown/"), new File("src/test/resources/org/broadinstitute/barclay/help/expected/markdown"), "yml", "md"}
         };
     }
 
     @Test(dataProvider = "getDocGenTestParams")
-    public void testDocGenRoundTrip(final Class<?> docletClass, final File templatesFolder, final File expectedDir) throws IOException {
+    public void testDocGenRoundTrip(final Class<?> docletClass, final File templatesFolder, final File expectedDir,
+            final String indexFileExtension, final String outputFileExtension) throws IOException {
```

Comment:
It seem unnecessary to add an additional set of input templates and expected files just so we can test that the output extension works correctly. Can you just re-use the existing input templates and expected files, and just change the test to know what output extension to expect ? The contents should be the same no matter what the extension.